industry_benchmarks/utils/data_gathering.py

```diff
@@ -444,14 +445,224 @@ def gather_ligand_scores(
+    )
+    results_dir = (
+        pathlib.Path(result["unit_results"][proto_key]["outputs"]["nc"]["path"])
+        .resolve()
+        .parent
+    )
+    # if the dir doesn't exist, we should try and fix it
+    if not results_dir.is_dir():
+        print("Fixing path to results dir")
+        # Depending on the relative location to the result dir, we might have
+        # to fix a duplicate folder, see this post for more details
+        # https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/83#discussion_r1689003616
+        results_dir = remove_first_reversed_sequential_duplicate_from_path(
+            results_dir
+        )
+        # Now we should check if the dir exists
+        if not results_dir.is_dir():
+            print("Can't find results directory, skipping")
+            return None
+
+    # now check that all of the results files can be found in the folder
+    for f_name in [
+        "structural_analysis_data.npz",
+        "energy_replica_state.npz",
+        "simulation_real_time_analysis",
+        "info.yaml"
+    ]:
+        if not results_dir.joinpath(f_name).exists():
+            print(f"Can't find cleaned results files for {f_name}")
+            return None
+
+    return results_dir
+
+def get_transform_name(result: dict, alchemical_network: gufe.AlchemicalNetwork) -> str:
+    """
+    Get the name of this transformation, taking into account that the inputs might have accidentally been deleted.
+    """
+    # grab the gufe key of the chemical systems used in the inputs
+    unit_result = list(result["unit_results"].values())[0]
+    state_a_key = unit_result["inputs"]["stateA"][":gufe-key:"]
+    mapping_key = unit_result["inputs"]["ligandmapping"][":gufe-key:"]
+    # work out which system this is in the alchemical network
+    system_look_up = dict((str(node.key), node) for node in alchemical_network.nodes)
+    mapping_look_up = dict((str(edge.mapping.key), edge.mapping) for edge in alchemical_network.edges)
+    # build the transform
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in system_look_up[state_a_key].components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    ligmap = mapping_look_up[mapping_key]
+
+    name = f"{phase}_{ligmap.componentA.name}_{ligmap.componentB.name}"
+    return name
+
+
+def  process_results(results_folders: list[pathlib.Path], output_dir: pathlib.Path, alchemical_network: gufe.AlchemicalNetwork) -> list[str]:
+    """
+    Loop over the results folders extracting the required information and moving it to the output folder.
+
+    Returns
+    -------
+        failed_results: list[str]
+        A list of edge names which have missing results
+    """
+    # workout the expected number of results
+    # assuming 3 repeats of each solvent and complex transformation
+    missing_results = []
+    all_results = {}
+    for edge in alchemical_network.edges:
+        for phase in ["complex", "solvent"]:
```

Comment:
The `AlchemicalNetwork` should have edges for both the complex and the solvent, i.e. looping over the two phases for every edges may create everything twice (e.g. 11 edges in the ligand network would be 22 edges in the alchemical network).

Reply:
Ahh this is left over from using the ligand network in the first draft, great catch!